create ijepa modelcard (ref : PR  #36979 ). (#39354)
huggingface/transformers: docs/source/en/model_doc/ijepa.md

This paper demonstrates an approach for learning highly semantic image representations without relying on hand-crafted data-augmentations. We introduce the Image- based Joint-Embedding Predictive Architecture (I-JEPA), a non-generative approach for self-supervised learning from images. The idea behind I-JEPA is simple: from a single context block, predict the representations of various target blocks in the same image. A core design choice to guide I-JEPA towards producing semantic representations is the masking strategy; specifically, it is crucial to (a) sample tar- get blocks with sufficiently large scale (semantic), and to (b) use a sufficiently informative (spatially distributed) context block. Empirically, when combined with Vision Transform- ers, we find I-JEPA to be highly scalable. For instance, we train a ViT-Huge/14 on ImageNet using 16 A100 GPUs in under 72 hours to achieve strong downstream performance across a wide range of tasks, from linear classification to object counting and depth prediction.
The example below demonstrates how to extract image features with [`Pipeline`] or the [`AutoModel`] class.

> <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/model_doc/ijepa_architecture.jpg"

Reviewer: Lets keep the image

)
features = feature_extractor("https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/pipeline-cat-chonk.jpeg", return_tensors=True)  

print(f"Feature shape: {features.shape}")

```

</hfoption>
<hfoption id="AutoModel">

```py
import requests
import torch
from PIL import Image
from torch.nn.functional import cosine_similarity
from transformers import AutoModel, AutoProcessor  

url_1 = "http://images.cocodataset.org/val2017/000000039769.jpg"  
url_2 = "http://images.cocodataset.org/val2017/000000219578.jpg"
image_1 = Image.open(requests.get(url_1, stream=True).raw)
image_2 = Image.open(requests.get(url_2, stream=True).raw)

processor = AutoProcessor.from_pretrained("facebook/ijepa_vith14_1k")  
model = AutoModel.from_pretrained("facebook/ijepa_vith14_1k", torch_dtype="auto", attn_implementation="sdpa")  

def infer(image):  
    inputs = processor(image, return_tensors="pt")  
    outputs = model(**inputs)  
    return outputs.last_hidden_state.mean(dim=1)  

embed_1 = infer(image_1)  
embed_2 = infer(image_2)  

similarity = cosine_similarity(embed_1, embed_2)  
print(similarity)
```
</hfoption>
</hfoptions>

## Quantization

> The following example demonstrates how to quantize I-JEPA models using bitsandbytes for 4-bit and 8-bit quantization to reduce model size and memory usage while maintaining performance. More quantization methods are discussed [here](https://huggingface.co/docs/transformers/v4.53.2/quantization/overview).

Reviewer: ```suggestion
Quantization reduces the memory burden of large models by representing the weights in a lower precision. Refer to the [Quantization](../quantization/overview) overview for more available quantization backends.

The example below uses [bitsandbytes](../quantization/bitsandbytes) to only quantize the weights to 4-bits.
```

> <hfoptions>
> <hfoption id="Quantization">

Reviewer: ```suggestion
```

> ```py

Reviewer: ```py
import torch
from transformers import BitsAndBytesConfig, AutoModel, AutoProcessor
from datasets import load_dataset

quantization_config = BitsAndBytesConfig(
    load_in_4bit=True,
    bnb_4bit_quant_type="nf4",
    bnb_4bit_compute_dtype=torch.bfloat16,
    bnb_4bit_use_double_quant=True,
)

url_1 = "http://images.cocodataset.org/val2017/000000039769.jpg"
url_2 = "http://images.cocodataset.org/val2017/000000219578.jpg"
image_1 = Image.open(requests.get(url_1, stream=True).raw)
image_2 = Image.open(requests.get(url_2, stream=True).raw)

processor = AutoProcessor.from_pretrained("facebook/ijepa_vith14_1k")
model = AutoModel.from_pretrained("facebook/ijepa_vith14_1k", quantization_config=quantization_config, torch_dtype="auto", attn_implementation="sdpa")

def infer(image):
    inputs = processor(image, return_tensors="pt")
    outputs = model(**inputs)
    return outputs.last_hidden_state.mean(dim=1)

embed_1 = infer(image_1)
embed_2 = infer(image_2)

similarity = cosine_similarity(embed_1, embed_2)
print(similarity)
```

import torch
from transformers import BitsAndBytesConfig, AutoModel, AutoProcessor
from datasets import load_dataset

quantization_config = BitsAndBytesConfig(
    load_in_4bit=True,
    bnb_4bit_quant_type="nf4",
    bnb_4bit_compute_dtype=torch.bfloat16,
    bnb_4bit_use_double_quant=True,
)

model = AutoModel.from_pretrained(
    "facebook/ijepa_vith14_1k",
    quantization_config=quantization_config,
    torch_dtype=torch.bfloat16,
    device_map="auto"
)

processor = AutoProcessor.from_pretrained("facebook/ijepa_vith14_1k")

dataset = load_dataset("huggingface/cats-image")
image = dataset["test"][0]["image"]
inputs = processor(image, return_tensors="pt")

with torch.no_grad():
    outputs = model(**inputs)
    features = outputs.last_hidden_state.mean(dim=1)

print(f"Features shape: {features.shape}")
print(f"Memory usage: {model.get_memory_footprint() / 1e9:.2f} GB")
```

> </hfoption>
> </hfoptions>

Reviewer: ```suggestion
```

    outputs = model(**inputs)
    return outputs.last_hidden_state.mean(dim=1)

embed_1 = infer(image_1)
embed_2 = infer(image_2)

similarity = cosine_similarity(embed_1, embed_2)
print(f"Similarity: {similarity.item():.4f}")
```

</hfoption>
</hfoptions>

## Notes

- I-JEPA is designed specifically for feature extraction and doesn't generate pixel-level predictions like traditional autoencoders.
- The model uses a masking strategy that samples target blocks with large semantic scale while using spatially distributed context blocks.
- I-JEPA works best with Vision Transformers and scales efficiently to large model sizes.

    ```py
    # Example of using I-JEPA for downstream classification task
    from transformers import IJepaForImageClassification, AutoProcessor

    model = IJepaForImageClassification.from_pretrained("facebook/ijepa_vith14_1k")
    processor = AutoProcessor.from_pretrained("facebook/ijepa_vith14_1k")
    
    inputs = processor(images, return_tensors="pt")
    outputs = model(**inputs)
    ```

> ## Resources

Reviewer: Can remove the `## Resources` section as well
